pyproject.toml

```diff
@@ -50,9 +50,15 @@ pytest-operator = "^0.23.0"
 juju = "^2" # Latest juju 2
 psycopg2 = {version = "^2.9.5", extras = ["binary"]}
 
+[tool.poetry.group.ha_charm]
+optional = true
+
 [tool.poetry.group.ha_charm.dependencies]
 ops = "^2.0"
-tenacity = "^8.2.1"
+tenacity = "^8.2.2"
+
+[tool.poetry.group.relation_charm]
+optional = true
```

Comment:
Is this related to the `Optional` type imported by the data_interfaces library?

Reply:
> Is this related to the `Optional` type imported by the data_interfaces library?

It's unrelated. It is a [configuration for poetry](https://python-poetry.org/docs/managing-dependencies/#optional-groups), I've added it since those groups are unrelated to the main charm itself or its test suites.